sound/soc/sof/sof-client.c

```diff
@@ -126,6 +127,25 @@ static inline int sof_register_ipc_msg_injector(struct snd_sof_dev *sdev)
 static inline void sof_unregister_ipc_msg_injector(struct snd_sof_dev *sdev) {}
 #endif /* CONFIG_SND_SOC_SOF_DEBUG_IPC_MSG_INJECTOR */
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_KERNEL_INJECTOR)
+static int sof_register_ipc_kernel_injector(struct snd_sof_dev *sdev)
+{
```

Comment:
```c
if (sdev->pdata->ipc_type != SOF_IPC)
	return 0;
```
To block the fake DSP to be instantiated when IPC4 is used?

Reply:
Yea, IPC4 has no RX handlers last I looked, I will add it once there is something to actually parse the data

Reply:
can you update the PR to not create the aux device in case of IPC4? We can enable the fake DSP when it is supporting IPC4.

Reply:
I still think that the aux device should not be created in the first place
```c
if (sdev->pdata->ipc_type != SOF_IPC)
		return 0;
```
and the injector driver then can be free form IPC version check for now as it is only going to probe with a known IPC version.